orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/RedisQueue.kt

```diff
@@ -229,6 +244,13 @@ class RedisQueue(
       } else {
         try {
           val message = convertToMessage(json, mapper)
+
+          // TODO: AttemptsAttribute could replace `attemptsKey`
+          message.setAttribute(
```

Comment:
Thoughts on moving away from the `attemptsKey` to an attribute of the message itself?

It would solve issues where a message is re-pushed to the queue. Currently those messages receive a new `UUID` and therefore lose any previous attempts tracking. 

This was the crux of the problem I was looking to solve with this PR so happy to discuss further in person.